src/Type/Generic/TemplateTypeVariance.php

```diff
@@ -85,14 +85,18 @@ public function compose(self $other): self
 
 		if ($this->covariant()) {
 			if ($other->contravariant()) {
-				return self::createCovariant();
+				return self::createContravariant();
 			}
 			if ($other->covariant()) {
 				return self::createCovariant();
 			}
 			return self::createInvariant();
 		}
 
+		if ($this->invariant()) {
```

Comment:
This change has broken some things. But is this change wrong?

Reply:
> this has also been a bug in PHPStan since 2019

Can you please show the bug on an existing code on phpstan.org/try? Your example uses `@template-contravariant` which is a new feature (so it couldn't be a bug in 2019).

Reply:
Sure thing. Here's a similar example with `@template-covariant`: https://phpstan.org/r/144953ee-3dfd-4606-a558-b373309ef9c5

It should have given the following error, but it currently passes: `Template type K is declared as covariant, but occurs in invariant position in parameter a of method C::a().`

Reply:
Any updates on this @ondrejmirtes?